typing/ctype.ml

```diff
@@ -279,7 +297,7 @@ type unification_environment =
         equations_generation : equations_generation;
         assume_injective : bool;
         allow_recursive_equations : bool;
-        gadt_equations_level : int;
+        equations_scope : int;
         unify_eq_set : TypePairs.t; }
```

Comment:
I wonder what the code would look like if you had `env : Pattern_environment.t` here. It may look clumsier because more indirections have to be followed, or it may look nicer because it is a conceptual simplification.

Reply:
The suggestion has actually worked very well,
both for simplifying `unification_environment` and the coercion into it.